microsoft/testsuites/kata_conformance/kata_conformance_tests_tool.py

```diff
@@ -0,0 +1,270 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import json
+import os
+from dataclasses import dataclass
+from pathlib import Path, PurePath
+from typing import Any, List, Type
+
+from kubernetes import client, config, utils
+
+from lisa import Environment, notifier
+from lisa.executable import Tool
+from lisa.messages import SubTestMessage, TestStatus, create_test_result_message
+from lisa.testsuite import TestResult
+from lisa.tools import Echo, Git, Go, Make
+from lisa.util import LisaException
+
+
+@dataclass
+class KataConformanceTestResult:
+    name: str = ""
+    status: TestStatus = TestStatus.QUEUED
+
+
+class KataConformanceTests(Tool):
+    # These tests take some time to finish executing. The default
+    # timeout of 600 is not sufficient.
+    TIME_OUT = 18000
+
+    cmd_path: PurePath
+    cmd_args: str
+    repo_root: PurePath
+    kube_config_path: PurePath
+
+    kubernetes_repo = "https://github.com/kubernetes/kubernetes.git"
+    kata_container_tests_repo = "https://github.com/kata-containers/tests.git"
+    deps = ["git", "make"]
+
+    @property
+    def command(self) -> str:
+        return str(self.cmd_path)
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    @property
+    def dependencies(self) -> List[Type[Tool]]:
+        return [Git, Make, Go]
+
+    def run_tests(
+        self, test_result: TestResult, environment: Environment, failure_logs_path: Path
+    ) -> None:
+        echo = self.node.tools[Echo]
+        original_path = echo.run(
```

Comment:
The go path part should be handled by go itself. The Go tool should provide a method like `go_run`.

Reply:
The PATH should be maintained in go tool, because it's part of the go tool. The go tool can provide a `go_run` method. The method can use the PATH as the env.

Reply:
We need to set PATH everytime when we execute command.

Here we will not use GO command explicitly to run or build something. 

When we perform MAKE on Kubernetes repo, it expects GO command in PATH. That is why i am calling _install_dep() first and then building Kubernetes with MAKE in _install() method.

I will create new method to retrieve PATH with go tool.

Reply:
What I mean is the logic should be in the `go.py` instead of the upper layer tool. So the Go tool in `go.py` is easier to use.